packages/dependency-extraction-webpack-plugin/lib/util.js

```diff
@@ -42,6 +42,9 @@ function defaultRequestToExternal( request ) {
 
 		case 'react-dom':
 			return 'ReactDOM';
+
+		case 'react/jsx-runtime':
+			return 'ReactJSXRuntime';
```

Comment:
The idea is to use an external script and a global for the jsx imports.

Reply:
It is very similar to how I integrated React Fast Refresh, so it makes perfect sense to replicate it as the use case is nearly identical.